lib/components/wallet_connect/approve_token_transfer.dart

```diff
@@ -0,0 +1,257 @@
+// ignore_for_file: prefer_const_constructors
+
+import 'package:flutter/material.dart';
+import 'package:persistent_bottom_nav_bar_v2/persistent_bottom_nav_bar_v2.dart';
+import 'package:qubic_wallet/components/wallet_connect/amount_value_header.dart';
+import 'package:qubic_wallet/di.dart';
+import 'package:qubic_wallet/flutter_flow/theme_paddings.dart';
+import 'package:qubic_wallet/helpers/global_snack_bar.dart';
+import 'package:qubic_wallet/helpers/re_auth_dialog.dart';
+import 'package:qubic_wallet/helpers/sendTransaction.dart';
+import 'package:qubic_wallet/helpers/target_tick.dart';
+import 'package:qubic_wallet/l10n/l10n.dart';
+import 'package:qubic_wallet/models/wallet_connect/approve_token_transfer_result.dart';
+import 'package:qubic_wallet/resources/qubic_li.dart';
+import 'package:qubic_wallet/services/wallet_connect_service.dart';
+import 'package:qubic_wallet/stores/application_store.dart';
+import 'package:qubic_wallet/styles/edge_insets.dart';
+import 'package:qubic_wallet/styles/text_styles.dart';
+import 'package:qubic_wallet/styles/themed_controls.dart';
+import 'package:walletconnect_flutter_v2/walletconnect_flutter_v2.dart';
+
+class ApproveTokenTransfer extends StatefulWidget {
+  final PairingMetadata? pairingMetadata;
+  final String? fromID;
+  final String? fromName;
+  final int amount;
+  final String? toID;
+  const ApproveTokenTransfer(
+      {super.key,
+      required this.pairingMetadata,
+      required this.fromID,
+      required this.fromName,
+      required this.amount,
+      required this.toID});
+
+  @override
+  // ignore: library_private_types_in_public_api
+  _ApproveTokenTransferState createState() => _ApproveTokenTransferState();
+}
+
+class _ApproveTokenTransferState extends State<ApproveTokenTransfer> {
+  final ApplicationStore appStore = getIt<ApplicationStore>();
+  final WalletConnectService wcService = getIt<WalletConnectService>();
```

Comment:
is wcService used in this class? remove if not

Reply:
Removed